### ISSUE TITLE
Specify default MySQL collation for new tables/columns

### ISSUE THREAD

**Issue by solokhind** [prisma/prisma#14179]
## Problem

With MySQL, when adding new models in the schema file and after migrating, prisma generates the SQL queries with a character set `utf8mb4` and collation `utf8mb4_unicode_ci`.

## Suggested solution

Do not specify the collations which will allow the tables to inherit the database's default collation and character set.

## Alternatives

Option to specify the default character set and collation in the schema file.

## Additional context

This has also been discussed here: [https://github.com/prisma/prisma/discussions/4743](https://github.com/prisma/prisma/discussions/4743)

**Comment by Pranav-Dakshina** [comment]
Any update on this issue?

**Comment by solokhind** [comment]
Still no updates unfortunately. Badly needed, I keep having to deal with this issue and I can't change the character set that's already in the production database. I have to manually deleted the default character collations in the generated mysql. I use the `--create-only` flag for that.

**Comment by Pranav-Dakshina** [comment]
I see Thanks @solokhind I guess I would have to do the same in the meantime.

**Comment by janpio** [comment]
Related: https://github.com/prisma/prisma/issues/12845

**Comment by yagomassiah** [comment]
This is a major problem also when baselining your database, you do a dp pull and suddenly the generated files mention a collation you never used. So 3 months in I realized my init_0 migrations is also wrong. 

I feel like the change here is, it should not specify the charset and collation at all unless is added to the schema. Any progress on this?

**Comment by yagomassiah** [comment]
@pimeys I saw that you classified this as feature but are we sure this is the case? it is also not a bug per se, but a feature that gets in the way of a normal workflow

By the way the solution is basically doing  `prisma migrate dev --create-only` always, so you can remove the hardset collation, for those of you that found this thread.  as @solokhind pointed out.

**Comment by janpio** [comment]
We currently do not support defining the collation, or setting a default, so for us this is new functionality which makes it a feature request @yagomassiah.

**Comment by yagomassiah** [comment]
@janpio makes sense, but my point here, instead of adding the feature of selecting the collation, why not just remove the hardset collation that the system is currently appending to create tables? later in the future of course you guys can add the feature. 

but the problem right know is the hardset utf8mb4_unicode_ci which I'm no sure where it comes from. So looks more like removing a broken feature. 

Thanks for taking the time to check this thread though!

**Comment by yagomassiah** [comment]
Great to hear @janpio,
But explicitly trying to guess the database collation instead of the default one seems more problematic. 

You're essentially forcing an alien database collation on anyone that is not on `utf8mb4_unicode_ci`, the problem with this approach is that it is a silent problem until it isn't. So introducing database conflicts to the users that are unaware of this until they need foreign keys on uuid's for instance. 

As the popularity of the migration tool grows, users can be introducing these conflicts for months before they realize it's already in production. 

```
-- CreateTable A 
CREATE TABLE `a` ( ); 
```
will yield you a table with a collation that defaults to your database settings. 

```
-- CreateTable B
CREATE TABLE `b` ( ) DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_unicode_ci;
``` 
will yield you a table with these collations, the moment you need to connect  A->B then you're going to realize what happened and you now have a problem on your hands. (not the end of the world though) 

but yeah, not sure is smart to let the users continuously add different collations than the previous tables.

**Comment by reubenjh** [comment]
Been dealing with this for about 1.5 years until i finally narrowed it down to a prisma issues today. Thought this was an issue with my default collation between different between my planetscale database branches, but no, its just anytime i run prisma push with a new implicit or explicit table being created, it will create it with a collation out of kilter with the rest of my db. 

My workflow now includes a script to explicitly set the collation to my target one (the db and table default) and i need to remember to run it if im making a schema change that needs it.

Agree this seems far more like a bug to be fixed than a feature request. Letting the db and table defaults decide which collation gets used seems the most intuitive approach.

**Comment by yagomassiah** [comment]
Hey @janpio is this fixed? after abandoning Prisma for a while I came back today and created a new project and used the migration tool and my new tables did not include the collation. You could tag this as closed if it can be confirmed.

**Comment by Cupcc** [comment]
Hi, can we get a bump on this thank you
